### PR TITLE
Modified the Polyfill to avoid rare issue with ".has" doesn't exist

### PR DESCRIPTION
**Pull request by maddada**
Had this issue with a customer's site after applying the polyfill for all customers, doesn't hurt to put null guards in these spots.

### CODE REVIEW

**Review by lenovo1996 — APPROVED** [review]

File: src/mutation_events.js [hunk 1]
```diff
@@ -128,11 +128,11 @@
 
   const observerOptions = {subtree: true, childList: true, attributes: true, attributeOldValue: true, characterData: true, characterDataOldValue: true};
   function enableMutationEventPolyfill(target) {
-    if (listeningNodes.has(target))
+    if (listeningNodes != null && listeningNodes.has != null && listeningNodes.has(target))
```
Comment (mfreed7) [inline src/mutation_events.js]:
```suggestion
    if (listeningNodes && listeningNodes.has && listeningNodes.has(target))
```

Comment (mfreed7) [reply src/mutation_events.js]: Ditto all cases below

**Review by mfreed7 — COMMENTED** [review]
Thanks for the PR! General side-question: how did it work for your use case? Was it an easy drop-in for mutation events? You're the first contact I've had with a user of the polyfill, so I'm curious.

The `has` method has been around for a while, but I don't see the harm in adding this. So LGTM, with a bit of cleanup requested.

Might even be better to add a helper function that does the `set && set.has && set.has(foo)` check, so the code isn't so repetitive?

**Comment by maddada** [comment]
It helped a lot, we managed to release a hotfix in a few hours which included this polyfill as we prepared the actual fix. Thank you for publishing it 🙏

We have a lot of customers on an older version of our product and the polyfill worked great to bring the product back to a working state for them after it got broken by the Chrome API deprecation. 

I had a few customers who were having an issue with the polyfill on the lines I edited (was giving an x is undefined error). I had to patch it for them and thought to contribute back. 

We already cleared up the issue with a new release, we were only using the deprecated APIs in a few places. We also introduced a process to warn us about any deprecated apis or libraries.

File: src/mutation_events.js [hunk 2]
```diff
@@ -143,11 +143,11 @@
   }
 
   function disableMutationEventPolyfill(target) {
-    if (!listeningNodes.has(target))
+    if (hasTarget(listeningNodes, target))
```
Comment (mfreed7) [inline src/mutation_events.js]: Should be `!hasTarget(...`

File: src/mutation_events.js [hunk 3]
```diff
@@ -143,11 +143,11 @@
...
       return;
     listeningNodes.delete(target);
     const rootElement = getRootElement(target);
-    if (!documentsToObservers.has(rootElement))
+    if (hasTarget(documentsToObservers, rootElement))
```
Comment (mfreed7) [inline src/mutation_events.js]: Should be `!hasTarget(...`

File: src/mutation_events.js [hunk 4]
```diff
@@ -214,4 +214,9 @@
   };
 
   console.log(`Mutation Events polyfill installed (native feature: ${nativeFeatureSupported ? "supported" : "not present"}).`);
+
+  // Helper function to avoid errors when has doesn't exist
```
Comment (mfreed7) [inline src/mutation_events.js]: Mind moving this up to around line 129 or so?

**Comment by mfreed7** [comment]
> It helped a lot, we managed to release a hotfix in a few hours which included this polyfill as we prepared the actual fix. Thank you for publishing it 🙏
> I had a few customers who were having an issue with the polyfill on the lines I edited (was giving an x is undefined error). I had to patch it for them and thought to contribute back.

Great, thanks for the feedback! Glad it helped. And thanks for contributing your change back.

File: src/mutation_events.js (hunk 2 above)
Comment (maddada) [reply src/mutation_events.js]: Apologies for the mistake, and thanks for catching it.

File: src/mutation_events.js (hunk 3 above)
Comment (maddada) [reply src/mutation_events.js]: ditto

File: src/mutation_events.js (hunk 4 above)
Comment (maddada) [reply src/mutation_events.js]: Done

**Review by mfreed7 — APPROVED** [review]

**Pull request merged**